docs/content/guides/dagster/deployment-options.mdx

```diff
@@ -0,0 +1,122 @@
+
+The default host for Dagit is `0.0.0.0` (your local machine), while the default port is `3000`. These can be custom set by adding variables to the command.
+
+```bash file=/guides/dagster/deployment-options/local_deploy.sh startafter=start_custom_command_marker endbefore=end_custom_command_marker
+DAGSTER_HOME=$(pwd) dagit -h 0.0.0.0 -p 3000
+```
+
+The lightweight setup comes with downsides, the largest one being the setup is not portable to other machines. If a pipeline requires other Python packages to be installed, they must be manually installed within a virtual environment on a new computer. Furthermore, additional steps are required if running schedules or sensors; in particular, the `dagster-daemon` needs to be running as well.
+
+The next two solutions aid with portability when additional complexity arises.
+
+## Using docker-compose
+
+<CodeReferenceLink filePath="examples/deploy_docker" />
+
+Any machine running Docker can run dockerized deployments. Code is be packaged into an image, which can be pulled into any other machine running Docker. For the local developer who would like to write and test pipelines, this option streamlines setup by only needing Docker to automatically install all other dependencies (like Python packages).
+
+Leveraging Docker also opens up a world of cloud-hosting Dagster on AWS or GCP by running the containers on a virtual machine, or better yet, a managed container service like ECS on AWS.
+
+For a Dagster deployment, three things are needed outside of the actual Python pipeline code:
+
+- `Dockerfile`: this serves as the instructions for building the actual image.
+- `docker-compose.yaml`: this file defines the multiple containers needed when running the application.
+- `dagster.yaml`: this file tells Dagster about the desired configuration settings, if not the defaults.
+
+### Dockerfile
+
+Running `docker build .` from the directory with the listed files would create the image, however this isn’t needed with the presence of the `build` section of the `docker-compose.yaml` file.
+
+The Dockerfile installs the relevant packages on top of a Python image, as well as sets the environment variables, host, and port for Dagit.
+
+The `ENTRYPOINT` is the exact same as the command for running Dagit locally, with the only difference being it runs automatically within a Docker container.
+
+### docker-compose.yaml
+
+The docker-compose file defines several services to be run in parallel on different containers.
+
+Four containers are launched from the file:
+
+- `docker_example_postgresql`: The Postgres container serves as the backend to storing logs and schedules.
+- `docker_example_pipelines`: This container loads and executes pipelines, used for both Dagit and the daemon.
```

Comment:
Something that's unclear to me: how does the pipeline code get packaged inside the container?  If the pipeline code depends on external packages, how do those get included as well?

In the past, I believe we've recommended using a setup.py and then `pip install`-ing it when building the container.  I can find some examples of that if it would be helpful.